Source/JavaScriptCore/llint/InPlaceInterpreter.asm

```diff
@@ -635,46 +633,534 @@ instructionLabel(_local_tee)
+    # pop data
+    popInt64(t1, t2)
+    # pop index
+    popInt64(t0, t2)
+    loadi [PM, MC], t2
+    addq t2, t0
+    # load memory location
+    loadp Wasm::Instance::m_cachedMemory[wasmInstance], t2
+    if ARM64E
+        removeArrayPtrTag t2
+    end
+    storeq t1, [t2, t0]
+
+    loadi 4[PM, MC], t0
+    advancePC(1)
+    advancePCByReg(t0)
+    advanceMC(8)
+    nextIPIntInstruction()
+
+instructionLabel(_f32_store_mem)
+    # f32.store
+    # pop data
+    popFloat32FT0()
+    # pop index
+    popInt32(t0, t2)
+    loadi [PM, MC], t2
+    addq t2, t0
+    # load memory location
+    loadp Wasm::Instance::m_cachedMemory[wasmInstance], t2
+    if ARM64E
+        removeArrayPtrTag t2
+    end
+    storef ft0, [t2, t0]
+
+    loadi 4[PM, MC], t0
+    advancePC(1)
+    advancePCByReg(t0)
+    advanceMC(8)
+    nextIPIntInstruction()
+
+instructionLabel(_f64_store_mem)
+    # f64.store
+    # pop data
+    popFloat64FT0()
+    # pop index
+    popInt32(t0, t2)
+    loadi [PM, MC], t2
+    addq t2, t0
+    # load memory location
+    loadp Wasm::Instance::m_cachedMemory[wasmInstance], t2
+    if ARM64E
+        removeArrayPtrTag t2
+    end
+    stored ft0, [t2, t0]
+
+    loadi 4[PM, MC], t0
+    advancePC(1)
+    advancePCByReg(t0)
+    advanceMC(8)
+    nextIPIntInstruction()
+
+instructionLabel(_i32_store8_mem)
+    # i32.store8
+    # pop data
+    popInt32(t1, t2)
+    # pop index
+    popInt32(t0, t2)
+    loadi [PM, MC], t2
+    addq t2, t0
+    # load memory location
+    loadp Wasm::Instance::m_cachedMemory[wasmInstance], t2
+    if ARM64E
+        removeArrayPtrTag t2
+    end
+    storeb t1, [t2, t0]
+
+    loadi 4[PM, MC], t0
+    advancePC(1)
+    advancePCByReg(t0)
+    advanceMC(8)
+    nextIPIntInstruction()
+
+instructionLabel(_i32_store16_mem)
+    # i32.store16
+    # pop data
+    popInt32(t1, t2)
+    # pop index
+    popInt32(t0, t2)
+    loadi [PM, MC], t2
+    addq t2, t0
+    # load memory location
+    loadp Wasm::Instance::m_cachedMemory[wasmInstance], t2
+    if ARM64E
+        removeArrayPtrTag t2
+    end
+    storeh t1, [t2, t0]
+
+    loadi 4[PM, MC], t0
+    advancePC(1)
+    advancePCByReg(t0)
+    advanceMC(8)
+    nextIPIntInstruction()
+    
+instructionLabel(_i64_store8_mem)
+    # i64.store8
+    # pop data
+    popInt64(t1, t2)
+    # pop index
+    popInt64(t0, t2)
+    loadi [PM, MC], t2
+    addq t2, t0
+    # load memory location
+    loadp Wasm::Instance::m_cachedMemory[wasmInstance], t2
+    if ARM64E
+        removeArrayPtrTag t2
+    end
+    storeb t1, [t2, t0]
+
+    loadi 4[PM, MC], t0
+    advancePC(1)
+    advancePCByReg(t0)
+    advanceMC(8)
+    nextIPIntInstruction()
+
+instructionLabel(_i64_store16_mem)
+    # i64.store16
+    # pop data
+    popInt64(t1, t2)
+    # pop index
+    popInt64(t0, t2)
+    loadi [PM, MC], t2
+    addq t2, t0
+    # load memory location
+    loadp Wasm::Instance::m_cachedMemory[wasmInstance], t2
+    if ARM64E
+        removeArrayPtrTag t2
+    end
```

Comment:
We should keep this pointer in some callee-save value, like WasmLLInt is doing.
This remove-array-ptr-tag causes crazy level of performance issue.
We are using memoryBase and boundsCheckingSize. And updating them only when necessary.